src/features/backup_manager.py

```
@@ -22,17 +22,47 @@ def get_commit_hash(project_path):
-    today = datetime.datetime.now().strftime("%Y-%m-%d")
-    ensure_dir(os.path.join(BACKUP_ROOT, today))
-    return os.path.join(BACKUP_ROOT, today, f"{nickname}-{commit_hash}.tar.gz")
+def get_valid_git_files(project_path):
+    """
+        Queries Git for all files that are not ignored by .gitignore
+        Returns a list of relative file paths.
+    """
+    try:
+        # 1. Get alll files tracked by Git
+        tracked = subprocess.run(
+            ["git", "ls-files"],
```

Ruff (0.16.1)

[error] Starting a process with a partial executable path

(S607)